tests/integration/base.py

```diff
@@ -362,9 +363,25 @@ def generate_and_settle(self):
                                                 stderr=subprocess.PIPE, universal_newlines=True)
                 self.fail('timed out waiting for networkd to settle down:\n%s\n%s\n%s' % (st, st_e, st_e2))
 
-        if subprocess.call(['nm-online', '--quiet', '--timeout=120', '--wait-for-startup']) != 0:
+        if subprocess.call(['nm-online', '--quiet', '--timeout=240', '--wait-for-startup']) != 0:
             self.fail('timed out waiting for NetworkManager to settle down')
 
+    def nm_online_full(self, iface, timeout=60):
+        '''Wait for NetworkManager connection to be completed (incl. IP4 & DHCP)'''
+
+        gi.require_version('NM', '1.0')
+        from gi.repository import NM
+        for t in range(timeout):
+            c = NM.Client.new(None)
+            con = c.get_device_by_iface(iface).get_active_connection()
+            if not con: self.fail('no active connection for %s by NM' % iface)
```

Comment:
Just a stylistic nit-pick: could we get this split into two lines? Since then it will fit better with the rest of the code.